.github/workflows/test.yml

```diff
@@ -93,30 +93,27 @@ jobs:
     smoke-test:
       name: smoke-test
       runs-on: ubuntu-22.04-github-hosted-16core
-      env:
-        TEST_REPO_DIR: test-repo
-  
+
       steps:
       - name: Checkout code
         uses: actions/checkout@v4 
         with:
-          path: ${{ env.TEST_REPO_DIR }}
+          submodules: recursive
           ref: ${{ github.event.pull_request.head.sha }}
       
       - name: Install Rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
           toolchain: nightly-2023-09-21
       
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+        
```

Comment:
I believe `actions-rust-lang/setup-rust-toolchain` actually configures this by default. https://github.com/actions-rust-lang/setup-rust-toolchain?tab=readme-ov-file#inputs

Reply:
Crazy think it's not working as intended maybe since all runs take ~ 17minutes to finish

Reply:
![image](https://github.com/matter-labs/foundry-zksync/assets/1564843/c9687a01-41a5-41eb-90ad-e542d0b40484)

That seems too small of a size to realistically cache anything